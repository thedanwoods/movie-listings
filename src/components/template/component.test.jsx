import React from 'react';
import { shallow } from 'enzyme';
import Template from './component';

const defaultProps = {
  tested: true,
  onClick: jest.fn(),
};

describe('<Template /> component', () => {
  it('renders without errors', () => {
    const template = shallow(<Template {...defaultProps} />);

    expect(() => template).not.toThrow();
    expect(template).toMatchSnapshot();
  });
});
