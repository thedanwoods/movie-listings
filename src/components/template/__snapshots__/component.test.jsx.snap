// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Template /> component renders without errors 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TemplateComponent
    onClick={[MockFunction]}
    tested={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          onClick={[MockFunction]}
          type="button"
        >
          This is a template component!
        </button>,
        <p>
          TESTED!
        </p>,
      ],
      "className": "template",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "This is a template component!",
          "onClick": [MockFunction],
          "type": "button",
        },
        "ref": null,
        "rendered": "This is a template component!",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "TESTED!",
        },
        "ref": null,
        "rendered": "TESTED!",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            onClick={[MockFunction]}
            type="button"
          >
            This is a template component!
          </button>,
          <p>
            TESTED!
          </p>,
        ],
        "className": "template",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "This is a template component!",
            "onClick": [MockFunction],
            "type": "button",
          },
          "ref": null,
          "rendered": "This is a template component!",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "TESTED!",
          },
          "ref": null,
          "rendered": "TESTED!",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
